### PR TITLE
display previous stages in the pipeline as well as the next ones

### PR DESCRIPTION
**Pull request by jonmoter**
Added a `Stage#previous_stages` method, and updated the view so it shows the previous stages as well as the next stages.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### Risks
- Level: Low

### CODE REVIEW

**Comment by grosser** [comment]
next PR: generate a graph and render it ? :D

**Comment by jonmoter** [comment]
@grosser could you try running this locally and confirming it looks okay?  When I try running Samson on my laptop, the ruby interpreter segfaults.

**Comment by grosser** [comment]
pushed a fix:

screen shot 2017-02-01 at 1 18 37 pm

**Comment by grosser** [comment]
try the grosser/24 branch ... does that help ?

File: plugins/pipelines/app/decorators/stage_decorator.rb [hunk 1]
```diff
@@ -19,6 +19,12 @@ def next_stages
     Stage.where(id: next_stage_ids).to_a
   end
 
+  def previous_stages
+    @previous_stages ||= project.stages.to_a.select do |stage|
+      stage.next_stage_ids.map(&:to_i).include? id
```
Comment (jonmoter) [inline plugins/pipelines/app/decorators/stage_decorator.rb]: @grosser is there some way to force `next_stage_ids` to be an array of ints?  It's pretty unintuitive that the method returns an array of strings.

**Comment by grosser** [comment]
yeah, this is messy ... ran into this a few times already :D
let's finish this PR and then do a cleanup PR

**Pull request merged**